packages/lib/storage/service.ts

```diff
@@ -42,19 +54,47 @@ type TGetFileResponse = {
   };
 };
 
-export const getFileFromS3 = async (fileKey: string) => {
-  const getObjectCommand = new GetObjectCommand({
-    Bucket: AWS_BUCKET_NAME,
-    Key: fileKey,
-  });
+// discriminated union
+type TGetSignedUrlResponse =
+  | { signedUrl: string; fileUrl: string; presignedFields: Object }
+  | {
+      signedUrl: string;
+      fileUrl: string;
+      signingData: {
+        signature: string;
+        timestamp: number;
+        uuid: string;
+      };
+    };
 
-  try {
-    const signedUrl = await getSignedUrl(s3Client, getObjectCommand, { expiresIn: 3600 });
+export const getFileFromS3 = (fileKey: string): Promise<string> => {
+  const [environmentId, accessType] = fileKey.split("/");
+  const expiresIn = accessType === "public" ? 60 * 60 : 10 * 60;
 
-    return signedUrl;
-  } catch (err) {
-    throw err;
-  }
+  const revalidateAfter = accessType === "public" ? expiresIn - 60 * 5 : expiresIn - 60 * 2;
+
+  return unstable_cache(
+    async () => {
+      const getObjectCommand = new GetObjectCommand({
+        Bucket: AWS_BUCKET_NAME,
+        Key: fileKey,
+      });
+
+      try {
+        return await getSignedUrl(s3Client, getObjectCommand, { expiresIn });
+      } catch (err) {
+        throw err;
+      }
+    },
+    [`getFileFromS3-${fileKey}`],
+    {
+      revalidate: revalidateAfter,
+      tags: [
+        storageCache.tag.byEnvironmentId(environmentId),
+        storageCache.tag.byAccessType(accessType as TAccessType),
```

Comment:
you introduces these revalidation tags but you never revalidate based on them. E.g. one reason to revalidate this cache would be because the files has been deleted. Then the best Tag would be
`storageCache.tag.byFileKey`
I'm not sure about the other tags. E.g. the tag `byAccessType("public")` would revalidate the caches of all files all over Formbricks Cloud if someone revalidates `public`. So I think we can remove this.
Environment id could make sense if we use it, but as we don't have listing services yet, I don't think we need it at the moment :-)